.github/workflows/continous-integration.yml

```diff
@@ -233,17 +236,19 @@ jobs:
         python-version: [ 3.7, 3.8, 3.9 ]
 
     steps:
-    - name: Run DataDog Agent
-      if: needs.changes.outputs.backend == 'true' && !((matrix.python-version == 3.8 || matrix.python-version == 3.9) && matrix.os == 'windows-2019')
```

Comment:
Because of the limitation we have of not being able to run pytest profiling on Windows, shouldn't we keep running datadog agent on windows even if we get less information?

Reply:
Unfortunately, we are not able to get any profiling information on Windows. If we enable it, the datadog agent will complain in the very beginning and [resulting in a silent failure](https://github.com/RasaHQ/rasa/runs/6453645890?check_suite_focus=true#step:16:35). 🙈

Reply:
I see, and this part that got removed cannot co-exist with the new step that you added?

Reply:
[The part](https://github.com/RasaHQ/rasa/blob/129bba0a64f43a0bfafdd41e508e3d4c02648dd2/.github/workflows/continous-integration.yml#L236-L241) that got deleted here is a docker setup for starting datadog agent. By the time we added this step, DataDog didn't provide an official solution for setting up an agent inside GitHub Actions. 

The official GitHub Actions were just released in Jan 2022. It is an elegant solution to properly start and stop an agent. Hence, we [replace](https://github.com/RasaHQ/rasa/blob/74fe422ac29ec9fec8c8479e9921b55487412347/.github/workflows/continous-integration.yml#L243-L250) the old docker commands with the official GitHub Actions.